src/depackers/arc_unpack.h

```diff
@@ -0,0 +1,119 @@
+/* Extended Module Player
+ * Copyright (C) 2021-2022 Alice Rowan <petrifiedrowan@gmail.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+/**
+ * Unpacker for ARC/ArcFS/Spark compressed streams.
+ * Report bugs to libxmp or to here: https://github.com/AliceLR/megazeuxtests
+ */
+
+#ifndef LIBXMP_ARC_UNPACK_H
+#define LIBXMP_ARC_UNPACK_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stddef.h> /* size_t */
+
+/* TODO libxmp should probably detect this in autoconf. */
+#if 0
+#define ARC_RESTRICT __restrict__
```

Comment:
Maybe this? (lodepng does this)
```c
#if (defined(__GNUC__) && (__GNUC__ > 3 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 1))) ||\
    (defined(_MSC_VER) && (_MSC_VER >= 1400)) || \
    (defined(__WATCOMC__) && (__WATCOMC__ >= 1250) && !defined(__cplusplus))
#define ARC_RESTRICT __restrict
#else
#define ARC_RESTRICT
#endif
```

Reply:
I assumed it would be more annoying to reliably detect than that. In that case, would it be worth adding a `LIBXMP_RESTRICT` or similar to src/common.h? Presumably some other things in libxmp could benefit from `restrict` as well.

Reply:
Oh yes.

Reply:
OK, I added `LIBXMP_RESTRICT` for this.